bycycle/plts/burst.py

```diff
@@ -0,0 +1,227 @@
+from scipy.stats import zscore
+
+import matplotlib.pyplot as plt
+
+from neurodsp.plts import plot_time_series, plot_bursts
+
+from bycycle.plts.cyclepoints import plot_cyclepoints
+from bycycle.plts.utils import apply_tlims, get_extrema
+
+###################################################################################################
+###################################################################################################
+
+def plot_burst_detect_summary(df, sig, fs, osc_kwargs, tlims=None, figsize=(15, 3),
+                              plot_only_result=False, burst_params=None, interp=True):
+    """
+    Create a plot to study how the cycle-by-cycle burst detection
+    algorithm determines bursting periods of a signal.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Dataframe output of :func:`~.compute_features`.
+    sig : 1d array
+        Time series to plot.
+    fs : float
+        Sampling rate, in Hz.
+    osc_kwargs : dict
+        Burst parameter keys and threshold value pairs, as defined in the 'burst_detection_kwargs'
+        argument of :func:`.compute_features`.
+    tlims : tuple of (float, float), optional, default: None
+        Start and stop times for plot.
+    figsize : tuple of (float, float), optional, default: (15, 3)
+        Size of each plot.
+    plot_only_result : bool, optional, default: False
+        Plot only the signal and bursts, excluding burst parameter plots.
+    burst_params : list of str, optional, default: None
+        The names of the ``df`` burst detection parameter columns. These names may differ from the
+        keys in ``osc_kwargs``. This is only needed for non-default burst parameters.
+    interp : bool
+        Interpolates points if true.
+
+
+    Notes
+    -----
+
+    - If plot_only_result = True: return a plot of the burst detection in which periods with bursts
+      are denoted in red.
+
+    - If plot_only_result = False: return a list of the fig handle followed by the 5 axes.
+
+    - In the top plot, the raw signal is plotted in black, and the red line indicates periods
+      defined as oscillatory bursts. The highlighted regions indicate when each burst requirement
+      was violated, color-coded consistently with the plots below.
+
+     - blue: amplitude_fraction_threshold
+     - red: amplitude_consistency_threshold
+     - yellow: period_consistency_threshold
+     - green: monotonicity_threshold
+
+    - Custom burst parameters may be defined in ``osc_kwargs`` and ``burst_params``, but the order
+      is expected to be the same for both arguments.
+
+    """
+
+    # Normalize signal
+    sig = zscore(sig)
+
+    # Determine time array and limits
+    times = np.arange(0, len(sig) / fs, 1 / fs)
+    tlims = (times[0], times[-1]) if tlims is None else tlims
+
+    # Determine if peak of troughs are the sides of an oscillation
+    _, side_e = get_extrema(df)
+
+    df_lim, sig_lim, times_lim = apply_tlims(df, sig, times, fs, tlims)
```

Comment:
I'm not totally clear on why you maintain two copies of things (like `df` and `df_lim`). It seems like if and once things are trimmed, this can be used across the board, right? 

For example below, I'm not sure why 
`plot_cyclepoints(df, sig, fs, ax=axes[0], tlims=tlims, plot_zerox=False,...)`
can't be:
`plot_cyclepoints(df_lim, sig_lim, fs, ax=axes[0], tlims=None, plot_zerox=False,...)`

And so on. If there is a reason this doesn't work, we might want to add a comment and note in this function as to the reasons and so on, and/or think about updating wtv would be needed such that this could work.